extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -271,19 +280,88 @@ void ConsumeWindowsEventLog::onTrigger(const std::shared_ptr<core::ProcessContex
     return;
   }
 
-  const auto flowFileCount = processQueue(session);
+  struct TimeDiff {
+    auto operator()() const {
```

Comment:
Since the scope is local and we want to be about to reliably print the duration as milliseconds, I suggest using `uint64_t` instead of `auto` for the return type and avoid casting at the call sites.

I can also live with wrapping the call sites in `uint64_t{ timeDiff() }`, but I prefer an explicit return type for the sake of easy logging.

Reply:
What is there a difference between `(uint64_t)timeDiff())` and `uint64_t{ timeDiff() }`?

Reply:
This suggests that the underlying type of `milliseconds` is a signed type and a signed -> unsigned conversion is narrowing. Narrowing conversions are not allowed with direct/copy list-initialization.

I was wrong when suggesting `uint64_t`, because the underlying type is a signed integer type. Try `int64_t`, that will most likely work. The standard only specifies "signed integer type of at least 45 bits". https://en.cppreference.com/w/cpp/chrono/duration
Please keep the list initialization to prevent lossy conversion in the future if the underlying type changed to a wider one, like `int128_t`.